lib/min_heap.rb

```diff
@@ -48,24 +56,54 @@ def to_s
   # Space complexity: ?
   def empty?
     raise NotImplementedError, "Method not implemented yet..."
+    return true if @store == []
   end
 
   private
 
+  # def find_parent(index)
+  #   parent_index = (index - 1)/2
+  #   return parent_index
+  # end
+
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
   # Time complexity: ?
   # Space complexity: ?
   def heap_up(index)
+    return if index == 0
+    parent_index = (index - 1) / 2
+    return if @store[parent_index].key < @store[index].key
+    swap(index, parent_index)
+    heap_up(parent_index)
     
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
-    raise NotImplementedError, "Method not implemented yet..."
+    # raise NotImplementedError, "Method not implemented yet..."
+    left_child = index * 2 + 1
+    right_child = index * 2 + 2
+    if @store[left_child] == nil #at bottom of tree
+      return
+    end
+    if @store[right_child] == nil
+      if @store[index] > @store[left_child]
```

Comment:
You should be comparing the keys here.  

```suggestion
      if @store[index].key > @store[left_child].key
```